ui/app/src/state/epics/itemVersions.ts

```diff
@@ -85,14 +88,22 @@ export default [
         )
       )
     ),
-  (action$) =>
+  (action$, state$: StateObservable<GlobalState>) =>
     action$.pipe(
       ofType(revertContentComplete.type),
-      switchMap(({ payload }) => [
-        fetchItemVersions(),
-        showRevertItemSuccessNotification(),
-        reloadDetailedItem({ path: payload.path })
-      ])
+      withLatestFrom(state$),
+      switchMap(([{ payload }, state]) => {
+        if (payload.path === state.preview.guest.path) {
+          getHostToGuestBus().next({ type: RELOAD_REQUEST });
+        }
```

Comment:
This logic applies to next too? Isn't this all for legacy? In which case, the host to guest bus doesn't do anything...

Reply:
This is for legacy but this is needed, now legacy LegacyConcierge is updating the preview.guest, also it is subscribe to RELOAD_REQUEST to call "CStudioAuthoring.Operations.refreshPreview()"